modal/_container_entrypoint.py

```diff
@@ -176,49 +176,53 @@ def get_max_inputs_to_fetch(self):
 
         return math.ceil(RTT_S / max(self.get_average_call_time(), 1e-6))
 
-    async def _generate_inputs(
-        self,
-    ) -> AsyncIterator[tuple[str, api_pb2.FunctionInput]]:
+    async def _generate_inputs(self) -> AsyncIterator[tuple[str, api_pb2.FunctionInput]]:
         request = api_pb2.FunctionGetInputsRequest(function_id=self.function_id)
         eof_received = False
         iteration = 0
         while not eof_received:
             request.average_call_time = self.get_average_call_time()
             request.max_values = self.get_max_inputs_to_fetch()  # Deprecated; remove.
+            request.input_concurrency = self.semaphore.maxsize
 
+            # If number of active inputs is at max queue size, this will block.
+            await self._acquire_semaphore()
```

Comment:
let's use a context manager for this – the documentation actually even mentions that it's the preferred way: https://docs.python.org/3/library/asyncio-sync.html#semaphore

Reply:
It's not possible to use a context manager without a huge refactor because the acquire and release happen in different functions.

Reply:
does it have to? it seems a bit risky to me that acquire and release are not 1:1